src/vmm/src/device_manager/persist.rs

```diff
@@ -78,6 +78,16 @@ pub struct ConnectedVsockState {
     pub mmio_slot: MMIODeviceInfo,
 }
 
+#[cfg(target_arch = "aarch64")]
+#[derive(Versionize)]
+/// Holds the state of a legacy device connected to the MMIO space.
+pub struct ConnectedLegacyState {
```

Comment:
Would it make sense to also save `DeviceType`? Using the `device_id` to identify the device type works for now but seems fragile.